### PR TITLE
feat!(content-manager): remove the ability to disable draft&publish

### LINKED ISSUE #16645: Husky broken on windows
### Required System information

- Node.js version: v18.9.1
- NPM version: 8.19.1
- Strapi version: not relevant since it is a repo issue
- Database: not relevant since it is a repo issue
- Operating system: windows 10
- Is your project Javascript or Typescript: not relevant since it is a repo issue

### Describe the bug

Husky pre commit hook broken causing me to not be able to comit on windows unless I reomve the pre-commit hook

### Steps to reproduce the behavior

1. be on windows
2. make a change to a file
3. try to commit to strapi 
4. get following error 
### Expected behavior

for husky to not error and for me to be able to comit

### Screenshots

b9e1293f60e525036c939095b8731844

### PR DESCRIPTION
**Pull request by alexandrebodin**
### What does it do?

Remove D&P triggering. D&P will always be active and the UX will be improved to allow a one step publish process

### Why is it needed?

Describe the issue you are solving.

### How to test it?

Provide information about the environment and the path to verify the behaviour.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

**Comment by joshuaellis** [comment]
Do we need to do this change now? something like this at 123 line changes especially effecting so many changes is going to cause _huge_ merge conflicts we're going to struggle with i think 🤔

**Comment by Marc-Roig** [comment]
At the moment , if I am not mistaken, the UI Edit View rely on the schema D&P flag to know if it's enabled or not, and then show the publish button or not.

Is the plan to always show draft and publish enabled in the content manager?

**Comment by joshuaellis** [comment]
> Is the plan to always show draft and publish enabled in the content manager

yes, it's always enabled

**Comment by alexandrebodin** [comment]
What kind of conflicts would you be expecting ? we don't have any plans to change D&P logic in v4 afaik. 

Keeping all of this around in v5 means we need to implement logic that we know we will remove in the docService, the contentApi etc this would be wasting quite some time

**Comment by joshuaellis** [comment]
> What kind of conflicts would you be expecting ? we don't have any plans to change D&P logic in v4 afaik.
> 
> Keeping all of this around in v5 means we need to implement logic that we know we will remove in the docService, the contentApi etc this would be wasting quite some time

I'm only looking at the number of files changes and the type of files – will Git be able to see the diff etc. But if you need to remove this because it reduces to the work to deliver the DocumentService then we should do it ofc 👍🏼

**Comment by joshuaellis** [comment]
Actually, another reason – these files aren't in TS yet – won't that be an issue? 🤔 I think for the FE it could be quite big, but you might not need to touch the FE ones now and i'll do it during the d&p work?

**Comment by alexandrebodin** [comment]
> Actually, another reason – these files aren't in TS yet – won't that be an issue? 🤔 I think for the FE it could be quite big, but you might not need to touch the FE ones now and i'll do it during the d&p work?

I don't think I can leave the FE alone or everything will be broken :/ I'll manage the conflicts myself if there are some
and otherwise I'll just redo it once we merge the ts migrations.

File: packages/core/content-manager/server/services/utils/populate.js
Comment (Marc-Roig) [inline packages/core/content-manager/server/services/utils/populate.js:171]: Have not tried, but I wonder what happens if the relation targets another one inside a component.
Components will not have the published_at attribute right?

I will try later during the day what happens if I try to publish an entry that has a relation targeting a compo.

Comment (alexandrebodin) [reply packages/core/content-manager/server/services/utils/populate.js:171]: Hum 🤔  you can't have a relation targeting a component

Comment (Marc-Roig) [reply packages/core/content-manager/server/services/utils/populate.js:171]: oh, we don't ? never mind then :D I thought we supported relations inside components

**Review by Marc-Roig — APPROVED** [review]

**Comment by alexandrebodin** [comment]
I'll wait for @joshuaellis  review before merging. If you both approve while I'm off go ahead and merge without me

**Review by joshuaellis — APPROVED** [review]
FE changes look minimal 💃🏼 lets get that conflict resolved...

**Pull request merged**
